Source/Core/Core/PowerPC/Jit64/Jit_LoadStore.cpp

```diff
@@ -318,38 +318,46 @@ void Jit64::dcbz(UGeckoInstruction inst)
 	int a = inst.RA;
 	int b = inst.RB;
 
-	u32 mem_mask = Memory::ADDR_MASK_HW_ACCESS;
-
-	// The following masks the region used by the GC/Wii virtual memory lib
-	mem_mask |= Memory::ADDR_MASK_MEM1;
-
 	MOV(32, R(RSCRATCH), gpr.R(b));
 	if (a)
 		ADD(32, R(RSCRATCH), gpr.R(a));
-	AND(32, R(RSCRATCH), Imm32(~31));
-	TEST(32, R(RSCRATCH), Imm32(mem_mask));
-	FixupBranch slow = J_CC(CC_NZ, true);
 
-	// Should this code ever run? I can't find any games that use DCBZ on non-physical addresses, but
-	// supposedly there are, at least for some MMU titles. Let's be careful and support it to be sure.
+	if (!UReg_MSR(MSR).DR)
```

Comment:
Compiling "twice"; we only execute the compiled code if DR is consistent between when the code is compiled and when it runs.  (We recompile on a mismatch.)